src/main/kotlin/no/nav/klage/dokument/service/DokumentUnderArbeidService.kt

```diff
@@ -675,7 +698,208 @@ class DokumentUnderArbeidService(
                         DocumentsChangedEvent.DocumentChanged(
                             id = dokumentUnderArbeid.id.toString(),
                             parentId = null,
-                            dokumentTypeId = dokumentUnderArbeid.dokumentType?.id,
+                            dokumentTypeId = dokumentUnderArbeid.dokumentType.id,
+                            tittel = dokumentUnderArbeid.name,
+                            isMarkertAvsluttet = dokumentUnderArbeid.erMarkertFerdig(),
+                        )
+                    ),
+                )
+            ),
+            behandlingId = behandling.id,
+            type = InternalEventType.DOCUMENTS_CHANGED,
+        )
+
+        return dokumentUnderArbeid
+    }
+
+    fun updateInngaaendeKanal(
+        behandlingId: UUID,
+        dokumentId: UUID,
+        inngaaendeKanal: InngaaendeKanal,
+        innloggetIdent: String
+    ): DokumentUnderArbeid {
+        val dokumentUnderArbeid = dokumentUnderArbeidRepository.findById(dokumentId).get()
+
+        val behandling = behandlingService.getBehandlingAndCheckLeseTilgangForPerson(behandlingId)
+
+        if (dokumentUnderArbeid.erMarkertFerdig()) {
+            throw DokumentValidationException("Kan ikke sette inngående kanal på et dokument som er ferdigstilt")
+        }
+
+        if (dokumentUnderArbeid.dokumentType != DokumentType.ANNEN_INNGAAENDE_POST) {
```

Comment:
Gjelder ikke inngående fra TR? F.eks. `!isInngaaende`?

Reply:
Egentlig ikke. Dette er oppdatering fra klienten, og for kjennelse fra TR skal ikke det tilbys. Der settes kanalen senere i løpet, i mapping mot KabalDocument.

Reply:
Hvis dette kallet blir gjort med et inngående fra TR så blir det ingen valideringsfeil da, eller valideres dette tidligere? Hvis denne metoden kun kan kalles for "Annen inngående post" så ville jeg renamet den litt til å være eksplisitt.

Reply:
Tja, det blir valideringsfeil her på 729 om det er kjennelse fra TR? Jeg syns vel egentlig at navnet er greit, jeg tenker det er primært at vi endrer inngående kanal, og at det er mer tilfeldig at det bare er relevant for "Annen inngående post". Men her kan en ha ulike mentale modeller :)

Reply:
OK. Men kan endre valideringsfeilen (teksten) litt iaf. Si noe om at man bare kan gjøre dette med `DokumentType.ANNEN_INNGAAENDE_POST`.

Reply:
Enig i det.